components/base/kinematicbase/ptgKinematics.go

```diff
@@ -133,11 +133,22 @@ func wrapWithPTGKinematics(
 		origin = originPIF.Pose()
 	}
 
+	// construct executionFrame
+	executionFrame, err := referenceframe.New2DMobileModelFrame(
+		b.Name().ShortName()+"ExecutionFrame",
+		planningFrame.DoF()[:3],
+		geometries[0],
```

Comment:
This is weird. Why just the first geometry? What are you doing with the DoF?

Reply:
you're right that this is weird. We now choose to not take geometry at all for the mobile frame. We also now define our own separate limits, although they _might be_ incorrect in this iteration.